website/src/app/kb/architecture/tech-stack/readme.mdx

```diff
@@ -0,0 +1,110 @@
+of the tech stack choices we made and why.
+
+## Control plane
+
+The control plane, which includes the
+[admin portal](/kb/architecture/core-components#admin-portal),
+[control plane API](/kb/architecture/core-components#control-plane-api), and
+[Policy Engine](/kb/architecture/core-components/#policy-engine), is built using
+Elixir and Phoenix.
+
+[Elixir](https://www.elixir-lang.org) is a functional programming language
+that's received lots of attention in recent years for its performance and
+scalability. It's built on top of the [Erlang VM](https://www.erlang.org/),
+which has a reputation for being fault-tolerant and highly concurrent.
+
+Together, these technologies power Firezone's realtime control plane API,
+allowing it to reliably handle thousands of policy decisions per second.
+
+## Data plane
+
+The data plane, which includes the Client, Gateway, and Relay, is built using
+Rust.
+
+[Rust](https://www.rust-lang.org) is a systems programming language that's known
+for its performance and safety. Not only does its memory safety guarantees
+prevent entire categories of security vulnerabilities, but it also has an
+outstanding [ecosystem](https://www.crates.io) of libraries and tools that make
+it a great choice for building performant network applications.
+
+### Client architecture
+
+Some parts of the macOS, iOS, and Android applications can't be built in Rust,
+and so a foreign function interface (FFI) is used to call into either Swift or
+Kotlin code appropriately.
+
+In general, we strive to keep the FFI architecture as simple as possible,
+leaving Rust-land only when absolutely required.
+
+Here's a high-level diagram of the various software components used in the
+Client applications and how to relate to each other on a per-platform basis:
+
+<Link
+  target="_blank"
+  href="/images/kb/architecture/tech-stack/client-architecture.svg"
+>
+  <Image
+    src="/images/kb/architecture/tech-stack/client-architecture.svg"
+    alt="Firezone client architecture diagram"
+    width={1200}
+    height={1200}
+  />
+</Link>
+
+## Ops and infrastructure
+
+Firezone uses the following tools for ops and infrastructure:
+
+| Category                | Tool/Service                  |
+| ----------------------- | ----------------------------- |
+| Cloud provider          | Google Cloud Platform         |
+| Source code management  | GitHub                        |
+| CI/CD                   | GitHub Actions                |
+| Monitoring and alerting | Google Cloud Monitoring       |
+| Logging                 | Google Cloud Logging          |
+| Persistence store       | Google Cloud SQL (PostgreSQL) |
+| Infrastructure as code  | Terraform                     |
+
+### Regional availability
+
+The Firezone-managed components are deployed globally across the following GCP
+zones for load balancing and latency optimization:
+
+| City                          | Region                 | Zones                    |
```

Comment:
I would not list zones here but, instead, give a link to `production/relays.tf`. So that we don't have keep this up to date.

Reply:
The accuracy here is not the important thing -- the target audience for this is probably a security manager or executive who just wants to see we have some regional availability. I'll link it for posterity though.

Reply:
Linked